### PR TITLE
Run make target to build docs in pr checks

### LINKED ISSUE #456: Add a CI check that builds the docs
### Is your feature request related to a specific problem?

It is possible for the docs to work just fine locally, but break in CI due to dependency issues

### Describe the solution you'd like

A CI check that builds the documentation to verify that it is valid and won't break on the `publish_docs` step

### Describe alternatives you've considered, if any

Nothing, as it is currently

### Additional context

This happens often due to optional dependencies that are actually needed for type information when importing fidesctl wholesale

### PR DESCRIPTION
**Pull request by earmenda**
Closes #456 

### Code Changes

* [x] Fix typo in file name reference in pr_checks workflow `pr_checks.yaml` -> `pr_checks.yml`
* [x] Created a new make target `docs-build-ci` that does not rebuild fidesctl
* [x] Created a new make target `docs-check` that builds the docs but doesn't serve them
* [x] Add `Docs` job to pr_checks that uses lightweight `make docs-check` target
* [x] Update publish docs job and serve-docs targets to also use optional dependencies using `[all]`
* [x] Remove publish docs pinned python version `3.9.0`

### Steps to Confirm

* [x] Manual testing of `make docs-serve`
* [x] Manual testing of `make docs-check`
* [x] Tested publishing `dev` docs within this PR - https://github.com/ethyca/fides/actions/runs/2163557590
* [x] Tested PR checks within this PR - https://github.com/ethyca/fides/actions/runs/2163562767

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* [x] Documentation Updated
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created

### Description Of Changes

This PR adds an additional pair of makefile targets that can generate and build our docs to verify that they are valid without serving them or deploying them. This is included as a standard CI check that runs when code changes

### CODE REVIEW

File: .github/workflows/publish_docs.yaml
Comment (ThomasLaPiana) [inline .github/workflows/publish_docs.yaml:31]: does this need quotes to work?

`pip install -e "fidesctl/[all]"`

Comment (earmenda) [reply .github/workflows/publish_docs.yaml]: Nope!

**Comment by earmenda** [comment]
This should be ready to merge unless there is any additional changes requested. I tested one more time that publishing works after merging the project restructuring. 

https://github.com/ethyca/fides/runs/6183058160?check_suite_focus=true

**Comment by ThomasLaPiana** [comment]
@earmenda I made a change here so that the PR check actually builds the docs site instead of just running the docs generation script. This will catch more errors and let us know if things are unbuildable, improper links, etc.

**Review by ThomasLaPiana — APPROVED** [review]
LGTM! Great QA change

**Pull request merged**
